rts/idris_stdfgn.c

```diff
@@ -259,3 +259,11 @@ VAL idris_getString(VM* vm, void* buffer) {
     free(sb);
     return str;
 }
+
+char* idris_currentDir() {
+   char cwd[1024];
+   if (getcwd(cwd, sizeof(cwd)) != NULL)
+     return cwd;
```

Comment:
Returning a pointer to a local buffer is wrong. It will kind of work until something else is called and that part of the callstack overwritten. 

Call MKSTR to create an idris string and return it as a VAL. Then set the return type in the FFI declaration on the IO side to IO (Raw String) and unpack it by matching on MkRaw.